mapviz_plugins/src/marker_plugin.cpp

```diff
@@ -203,7 +203,39 @@ namespace mapviz_plugins
         markerData.expire_time = ros::Time::now() + lifetime + ros::Duration(5);
       }
 
-      if (markerData.display_type == visualization_msgs::Marker::CYLINDER ||
+      if (markerData.display_type == visualization_msgs::Marker::ARROW)
+      {
+        StampedPoint point;
+        point.color = markerData.color;
+        point.orientation = orientation;
+
+        if (marker.points.empty())
```

Comment:
That works too. I just don't want to crash mapviz if someone sends it a bad message.